mm/2s2h/Enhancements/Cheats/TimeStop.cpp

```diff
@@ -0,0 +1,64 @@
+#include <libultraship/libultraship.h>
+#include "BenPort.h"
+#include "Enhancements/GameInteractor/GameInteractor.h"
+#include "Enhancements/Enhancements.h"
+
+extern "C" {
+#include <variables.h>
+#include <functions.h>
+}
+
+void RegisterTimeStopInTemples() {
+    GameInteractor::Instance->RegisterGameHook<GameInteractor::AfterRoomSceneCommands>([](s16 sceneId, s8 roomNum) {
+        uint8_t selectedOption = CVarGetInteger("gCheats.TempleTimeStop", 0);
+
+        switch (selectedOption) {
+            case TIME_STOP_TEMPLES_DUNGEONS:
+                switch (gPlayState->sceneId) {
+                    // Swamp + Ocean Spider House
+                    case SCENE_KINSTA1:
+                    case SCENE_KINDAN2:
+                    // Pirates' Fortress
+                    case SCENE_KAIZOKU:
+                    case SCENE_PIRATE:
+                    case SCENE_TORIDE:
+                    // Beneath the Well
+                    case SCENE_REDEAD:
+                    // Ancient Castle of Ikana + Igos's Lair
+                    case SCENE_CASTLE:
+                    case SCENE_IKNINSIDE:
+                    // Secret Shrine
+                    case SCENE_RANDOM:
+                        R_TIME_SPEED = 0;
+                        break;
+                    default:
+                        break;
+                }
```

Comment:
With the double switch statements and large blocks, it would be great to indicate that you are intentionally falling through with a comment.

```suggestion
                }
                // fallthrough
```